.github/workflows/lib-injection.yml

```diff
@@ -48,63 +48,117 @@ jobs:
+        # This results in the 'dd-lib-dotnet-init:<SHA>[-musl]' tag being overwritten by the latest build command, so the result is not a multiarch image
+
+        # Fix: The following lines create and publish a new manifest to overwrite the 'dd-lib-dotnet-init:<SHA>[-musl]' tag.
+        # In the new manifest we directly reference each platform-specific image, so the tag becomes a multiarch image.
+        
+        docker manifest create \
+          ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }} \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-amd64.outputs.digest}}" \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-arm64v8.outputs.digest}}"
+        docker manifest push ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }}
+
+        docker manifest create \
+          ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }}-musl \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-musl-amd64.outputs.digest}}" \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-musl-arm64v8.outputs.digest}}"
+        docker manifest push ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }}-musl
+
+  test:
+    needs:
+      - build-and-publish-init-image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        lib-injection-connection: ['network','uds']
+        lib-injection-use-admission-controller: ['', 'use-admission-controller']
+        weblog-variant: ['dd-lib-dotnet-init-test-app']
+        runtime: ['bullseye-slim','alpine']
+        include:
+          - runtime: 'bullseye-slim'
+            init-tag-suffix: ''
+          - runtime: 'alpine'
+            init-tag-suffix: '-musl'
+      fail-fast: false
+    env:
+      TEST_LIBRARY: dotnet
+      WEBLOG_VARIANT: ${{ matrix.weblog-variant }}
+      LIBRARY_INJECTION_CONNECTION: ${{ matrix.lib-injection-connection }}
+      LIBRARY_INJECTION_ADMISSION_CONTROLLER: ${{ matrix.lib-injection-use-admission-controller }}
+      DOCKER_REGISTRY_IMAGES_PATH: ghcr.io/datadog
+      DOCKER_IMAGE_TAG: ${{ github.sha }}${{ matrix.init-tag-suffix }}
+      DOCKER_IMAGE_WEBLOG_TAG: ${{ github.sha }}-${{ matrix.runtime }}
+      RUNTIME: ${{ matrix.runtime }}
+      BUILDX_PLATFORMS: linux/amd64
+    steps:
+      - name: lib-injection test runner
+        id: lib-injection-test-runner
+        uses: DataDog/system-tests/lib-injection/runner@69336117276b3ef31f2594d20cbd8cd8a7d26396 # zach/lib-injection-dotnet
+        with:
+          docker-registry: ghcr.io
+          docker-registry-username: ${{ github.repository_owner }}
+          docker-registry-password: ${{ secrets.GITHUB_TOKEN }}
+          test-script: ./lib-injection/run-manual-lib-injection.sh
```

Comment:
Bit of a shame that we only run the tests after we've already pushed a potentially faulty image.

I guess we don't have a choice though because IIRC you _have_ to push straight away when using buildx? 

Also, means we're not testing the arm64 images I assume 🤔

Reply:
You don't _have_ to push straight away when using buildx, we could remove the `push` argument. Perhaps we could build the images locally, invoke the tests (they run on the same machine), and then publish the images to the repo only after the tests pass? Think that's okay to pursue in a follow-up PR?

And yes, we're only testing amd64 at the moment. I'll have to check with the reliability to see how we might test the ARM64 image

Reply:
Indeed, I was thinking it's on GHCR so we don't care. 
But we could actually start a release after, and not realize the system tests are broken. It also means we should start being way more careful about a broken build when releasing (sometimes we put the sha to force using the last build even if it has partially failed, it could become riskier to do this now)

Reply:
> Perhaps we could build the images locally, invoke the tests (they run on the same machine), and then publish the images to the repo only after the tests pass? Think that's okay to pursue in a follow-up PR?

Sounds like it would be a good idea to me 👍